modules/shortcodes/instagram.php

```diff
@@ -262,6 +262,28 @@ function jetpack_shortcode_instagram( $atts ) {
 		return '';
 	}
 
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		$url_pattern = '#http(s?)://(www\.)?instagr(\.am|am\.com)/p/([^/?]+)#i';
+		preg_match( $url_pattern, $atts['url'], $matches );
+		if ( ! $matches ) {
+			return sprintf(
+				'<a href="%1$s" class="amp-wp-embed-fallback">%2$s</a>',
+				esc_url( $atts['url'] ),
```

Comment:
I missed that in the YouTube PR, but I think this can be simplified. See my PR in #14061 for an example.

Reply:
Ah, good point. How does https://github.com/Automattic/jetpack/pull/14053/commits/b03daedee340a3343799c99c921f7222db773741 look?